src/renderer/components/+workloads-overview/overview-statuses.tsx

```diff
@@ -9,73 +9,78 @@ import React from "react";
 import { observer } from "mobx-react";
 import { OverviewWorkloadStatus } from "./overview-workload-status";
 import { Link } from "react-router-dom";
-import { workloadStores } from "../+workloads";
 import type { NamespaceStore } from "../+namespaces/namespace-store/namespace.store";
 import type { KubeResource } from "../../../common/rbac";
 import { ResourceNames } from "../../utils/rbac";
-import { boundMethod } from "../../utils";
 import { workloadURL } from "../../../common/routes";
-import { isAllowedResource } from "../../../common/utils/allowed-resource";
 import { withInjectables } from "@ogre-tools/injectable-react";
 import namespaceStoreInjectable from "../+namespaces/namespace-store/namespace-store.injectable";
+import type { KubeObjectStore } from "../../../common/k8s-api/kube-object.store";
+import { podsStore } from "../+workloads-pods/pods.store";
+import { deploymentStore } from "../+workloads-deployments/deployments.store";
+import { daemonSetStore } from "../+workloads-daemonsets/daemonsets.store";
+import { statefulSetStore } from "../+workloads-statefulsets/statefulset.store";
+import { jobStore } from "../+workloads-jobs/job.store";
+import { cronJobStore } from "../+workloads-cronjobs/cronjob.store";
+import { replicaSetStore } from "../+workloads-replicasets/replicasets.store";
+import type { KubeObject } from "../../../common/k8s-api/kube-object";
+import type { IsAllowedResource } from "../../../common/utils/is-allowed-resource.injectable";
+import isAllowedResourceInjectable from "../../../common/utils/is-allowed-resource.injectable";
 
-const resources: KubeResource[] = [
-  "pods",
-  "deployments",
-  "statefulsets",
-  "daemonsets",
-  "replicasets",
-  "jobs",
-  "cronjobs",
-];
+const workloadStores = new Map<KubeResource, KubeObjectStore<KubeObject>>([
+  ["pods", podsStore],
+  ["deployments", deploymentStore],
+  ["daemonsets", daemonSetStore],
+  ["statefulsets", statefulSetStore],
+  ["replicasets", replicaSetStore],
+  ["jobs", jobStore],
+  ["cronjobs", cronJobStore],
+]);
+
+export interface OverviewStatusesProps {}
 
 interface Dependencies {
-  namespaceStore: NamespaceStore
+  namespaceStore: NamespaceStore;
+  isAllowedResource: IsAllowedResource;
 }
 
-@observer
-class NonInjectedOverviewStatuses extends React.Component<Dependencies> {
-  @boundMethod
-  renderWorkload(resource: KubeResource): React.ReactElement {
-    const store = workloadStores.get(resource);
+const NonInjectedOverviewStatuses = observer(({ namespaceStore, isAllowedResource }: Dependencies & OverviewStatusesProps) => {
```

Comment:
`namespaceStore` and `isAllowedResource` are wrong dependencies for this.

Should be `workloadEntries` which is `IComputedValue<{ resource: CHECK_TYPE, items: TYPE_OF_ITEM[], status:  TYPE_OF_STATUS, href: string }[]>`

That makes this component get all of it's business logic outside.

Reply:
Resolved with https://github.com/lensapp/lens/pull/4801/commits/a399e06aa433bd07905b57937eaaf013114dde48